feat(documentation): Update the contributing page to require signed commits (#36977)
DataDog/datadog-agent: docs/public/guidelines/contributing.md

  * If you think you've found an issue, please search the [Agent Troubleshooting][agent-troubleshooting] section to see if it's known.
  * If you’re still unsure about the issue, you may reach out to the [Datadog support](https://docs.datadoghq.com/help/) team with [a flare](https://docs.datadoghq.com/agent/troubleshooting/send_a_flare/) from your Agent.

>   * Finally, you can open a Github issue. Please add as much information as possible to help triaging the issue.

Reviewer: ```suggestion
  * Finally, you can open a Github issue. Add as much information as possible to help us triage the issue.
```

Your pull request must pass all CI tests before we will merge it. If you're seeing an error and don't think it's your fault, it may not be! [Join us on Slack](https://chat.datadoghq.com) or [send us an email](mailto:support@datadoghq.com), and together we'll get it sorted out.

### Merge protection

> To protect our repository, we enabled some checks. As an external contributor you are asked to sign our [Contributor Licence Agreement](https://cla.datadoghq.com/DataDog/datadog-agent) (CLA). You will have a message once your PR is opened to sign it.

Reviewer: ```suggestion
To protect our repository, we have enabled some checks. As an external contributor, you are required to sign our [Contributor License Agreement](https://cla.datadoghq.com/DataDog/datadog-agent) (CLA). You will receive a message once your PR is opened to sign the agreement.
```

> Besides, we require all commits to be signed before merging. To sign your commits you can follow [this procedure from Github](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

Reviewer: ```suggestion
We also require all commits to be signed before merging. To learn how to sign your commits, follow [this procedure from Github](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
```

> To protect our repository, we have enabled some checks. As an external contributor, you are required to sign our [Contributor License Agreement](https://cla.datadoghq.com/DataDog/datadog-agent) (CLA). You will receive a message once your PR is opened to sign the agreement.
> 
> We also require all commits to be signed before merging. To learn how to sign your commits, follow [this procedure from Github](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

Reviewer 2: ```suggestion
We have safety measures to protect our repository. As an external contributor, you are required to:

1. Sign our [Contributor License Agreement](https://cla.datadoghq.com/DataDog/datadog-agent) (CLA). You will receive a message once your PR is opened to sign the agreement.
2. Provide signed commits before merging. To learn how to sign your commits, follow [this procedure from Github](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
```

NIT: It seems a less odd to me like that and more explicit
